### PR TITLE
migrate image validation 4 rules to Docs Validation Lib

### PR DESCRIPTION
**Pull request by OsmondJiang**
[AB#268357](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/268357)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6209)

### CODE REVIEW

File: src/docfx/validation/ContentValidator.cs [hunk 1]
```diff
@@ -15,13 +16,32 @@ internal class ContentValidator
...
             _errorLog = log;
             _monikerProvider = monikerProvider;
             _publishUrlMap = publishUrlMap;
+            _links = new ConcurrentHashSet<SourceInfo<string>>();
+        }
+
+        public void ValidateImage(Document file, SourceInfo<string> link, string? altText)
+        {
+            // validate image link and altText here
+            if (_links.TryAdd(link) && TryGetValidationDocumentType(file.ContentType, file.Mime.Value, false, out var documentType))
```
Comment (yufeih) [inline src/docfx/validation/ContentValidator.cs:33]: What is this dedup here for?

File: src/docfx/lib/markdown/MarkdownEngine.cs [hunk 2]
```diff
@@ -296,6 +296,18 @@ private string GetLink(string path, MarkdownObject origin)
...
         }
 
+        private string GetImageLink(string path, MarkdownObject origin, string? altText)
+        {
+            return GetImageLink(new SourceInfo<string>(path, origin.GetSourceInfo()), altText);
+        }
+
+        private string GetImageLink(SourceInfo<string> href, string? altText)
+        {
+            _contentValidator.ValidateImage((Document)InclusionContext.File, href, altText);
```
Comment (yufeih) [inline src/docfx/lib/markdown/MarkdownEngine.cs:306]:
```suggestion
            _contentValidator.ValidateImageLink((Document)InclusionContext.File, href, altText);
```

File: docs/specs/validations/metadataContentValidations.yml [hunk 3]
```diff
@@ -1944,12 +1944,22 @@ inputs:
...
+    {"message_severity":"suggestion","log_item_type":"user","code":"image-alt-text-duplicated","message":"Alt text 'test alt text2' is duplicated. Within an article, alt text must be unique.","file":"a.md","line":15,"end_line":15,"column":1,"end_column":1}
+    {"message_severity":"suggestion","log_item_type":"user","code":"image-alt-text-duplicated","message":"Alt text 'test alt text2' is duplicated. Within an article, alt text must be unique.","file":"a.md","line":16,"end_line":16,"column":1,"end_column":1}
+    {"message_severity":"suggestion","log_item_type":"user","code":"image-alt-text-duplicated","message":"Alt text 'test alt text3' is duplicated. Within an article, alt text must be unique.","file":"a.md","line":18,"end_line":18,"column":11,"end_column":19}
+    {"message_severity":"suggestion","log_item_type":"user","code":"image-alt-text-duplicated","message":"Alt text 'test alt text3' is duplicated. Within an article, alt text must be unique.","file":"a.md","line":19,"end_line":19,"column":11,"end_column":19}
```
Comment (XiaofuHuang) [inline docs/specs/validations/metadataContentValidations.yml:1965]: Will we add the test case of the other three rules?

File: src/docfx/lib/html/HtmlToken.cs [hunk 4]
```diff
@@ -67,5 +67,18 @@ public void SetAttributeValue(string name, string value)
...
+        {
+            foreach (ref var attribute in Attributes.Span)
+            {
+                if (attribute.NameIs(name))
+                {
+                    return attribute.Value.ToString();
+                }
+            }
+
+            return "";
```
Comment (yufeih) [inline src/docfx/lib/html/HtmlToken.cs:81]: return `null` in case the attribute is missing
```suggestion
            return null;
```

**Pull request merged**